docs/src/prototypes/chatMessages/ControlMessages/index.tsx

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react'
+import {
+  Avatar,
+  Tree,
+  TreeTitle,
+  TreeItem,
+  FocusZoneMode,
+  FocusZoneDirection,
+  FocusZoneTabbableElements,
+  Chat,
+  Divider,
+  treeBehavior,
+  treeTitleBehavior,
+} from '@stardust-ui/react'
+import * as keyboardKey from 'keyboard-key'
+import { Accessibility } from 'src/lib/accessibility/types'
+import { ChatItemProps } from 'src/components/Chat/ChatItem'
+import { ShorthandCollection } from 'src/types'
+
+const janeAvatar = {
+  image: 'public/images/avatar/small/ade.jpg',
+  status: {
+    color: 'green',
+    icon: 'check',
+  },
+}
+
+const overridenTreeBehavior: Accessibility<any> = props => {
+  return {
+    ...treeBehavior(props),
+    focusZone: {
+      mode: FocusZoneMode.Embed,
+      props: {
+        shouldEnterInnerZone: event => keyboardKey.getCode(event) === keyboardKey.Enter,
+        direction: FocusZoneDirection.vertical,
+      },
+    },
+  }
+}
+
+const overridenTreeTitleBehavior: Accessibility<any> = props => {
+  return {
+    ...treeTitleBehavior(props),
+    focusZone: {
+      mode: FocusZoneMode.Embed,
+      props: {
+        handleTabKey: FocusZoneTabbableElements.all,
+        isCircularNavigation: true,
+        direction: FocusZoneDirection.vertical,
+      },
+    },
+  }
+}
+
+const ChatExample = () => {
+  const [open, setOpen] = React.useState(false)
+
+  const items: ShorthandCollection<ChatItemProps> = [
+    {
+      message: {
+        onKeyDown: e => {
+          const eventCode = keyboardKey.getCode(e)
+          if (eventCode === keyboardKey.Enter || eventCode === keyboardKey.Spacebar) {
+            setOpen(true)
+          }
+          if (eventCode === keyboardKey.Escape) {
+            // TODO very ugly check...
+            // if it comes from some tree title close the tree
+            if (
+              e.target.className.indexOf(TreeTitle.className) !== -1 ||
+              e.target.className.indexOf(TreeItem.className) !== -1
+            ) {
+              setOpen(false)
+            }
+          }
+        },
+        content: (
+          <Tree
+            data-is-focusable="true"
+            accessibility={overridenTreeBehavior}
+            items={[
+              {
+                onKeyDown: e => {
+                  // Without this override, the focus zone for the tree title is not working..
+                  console.log('Tree item 1 key down')
```

Comment:
The inner Focus zone doesn't work if I remove this event handler.. I tried just adding data-is-focusable, tabIndex: 0, but these two didn't help..